docs/development_decisions.md

```diff
@@ -4,9 +4,36 @@ The reasons why every tool used in the project was chosen are detailed in this f
+
+2. Active and frequent development.
+    It is crucial that the libraries used are regularly updated and patched to have up-to-date code that complies with the latest software requirements. If we chose any poorly maintained tool, we'd risk having to refactor that code to use a newer tool in the near future.
+
+3. Secure.
+    Nowadays it is imperative to always develop secure code and use secure libraries -the recent [Log4J](https://thehackernews.com/2021/12/extremely-critical-log4j-vulnerability.html) critical vulnerability is a good example of that-. Because of this, we will try to choose libraries that have not had notable security flaws in the past.
+
+4. Facilitates having the best practices of the programming language of choice.
+    We will try to choose libraries that are PEP compliant to achieve the goal of only developing quality code.
+
+### Task runner: **Invoke**
 [Invoke](https://www.pyinvoke.org/) is a task runner written in Python that allows defining your tasks in a pythonic way, while having an invocation style similar to *GNU Make*, which is very convenient for CLI-accustomed users. It is an open-source and well-maintained project and has an [extensive documentation](https://docs.pyinvoke.org/en/stable/) with lots of examples that make it very easy to start using it.  
 
-## Package and dependency manager: **Poetry**
+### Package and dependency manager: **Poetry**
 [Poetry](https://python-poetry.org/) is a packaging and dependency manager. It can resolve the multiple dependency constraints of a project if there's any possible solution and ease the task of working on a shared repository by ensuring that every developer is working with the same environment. It also supports building pure python wheels and publishing them.
-It also has a very easy-to-use command-line interface with tab completion and has a broad user-base within the Python community.
+It also has a very easy-to-use command-line interface with tab completion and has a broad user-base within the Python community.  
+
+### Test framework: **Pytest**
+When choosing a test framework we primarily considered [Unittest](https://docs.python.org/3/library/unittest.html) and [Pytest](https://docs.pytest.org/en/6.2.x/). We think that Pytest is more straightforward to use since Unittest needs more module importing and class-defining boilerplate, which can be a hassle to work with. Pytest has several plugins which can come in handy for later stages of the project, for example, one that allows users to generate HTTP reports using the *pytest-http* plugin. None of them have had remarkable security issues in the last years, so this point is a draw. As for the last point, we think that pytest would be a better fit because using a context manager to mock an object -as unittest does- can make the code less readable when having to mock different objects because of the nested `with` statements.  
```

Comment:
Really, not comparable, not by a long shot. One is an assertion library, the other a test runner with a few assertions and testing facilities for mocking and fixtures. Not comparable. You should try and check what other test runners are there, at least.